src/test/java/org/mskcc/cbio/oncokb/web/rest/SlackControllerIT.java

```diff
@@ -0,0 +1,647 @@
+import com.slack.api.model.block.SectionBlock;
+import com.slack.api.model.block.composition.TextObject;
+import com.slack.api.model.block.element.PlainTextInputElement;
+import com.slack.api.model.view.View;
+import com.slack.api.model.view.ViewState;
+import com.slack.api.util.json.GsonFactory;
+import com.slack.api.webhook.Payload;
+import io.github.jhipster.config.JHipsterProperties;
+import io.github.jhipster.security.RandomUtil;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Captor;
+import org.mockito.MockitoAnnotations;
+import org.mockito.Spy;
+import org.mskcc.cbio.oncokb.OncokbPublicApp;
+import org.mskcc.cbio.oncokb.config.application.ApplicationProperties;
+import org.mskcc.cbio.oncokb.config.application.EmailAddresses;
+import org.mskcc.cbio.oncokb.config.application.SlackProperties;
+import org.mskcc.cbio.oncokb.domain.Token;
+import org.mskcc.cbio.oncokb.domain.User;
+import org.mskcc.cbio.oncokb.domain.UserDetails;
+import org.mskcc.cbio.oncokb.domain.enumeration.LicenseType;
+import org.mskcc.cbio.oncokb.domain.enumeration.MailType;
+import org.mskcc.cbio.oncokb.repository.UserDetailsRepository;
+import org.mskcc.cbio.oncokb.repository.UserRepository;
+import org.mskcc.cbio.oncokb.service.*;
+import org.mskcc.cbio.oncokb.service.dto.UserDTO;
+import org.mskcc.cbio.oncokb.service.dto.useradditionalinfo.TrialAccount;
+import org.mskcc.cbio.oncokb.service.impl.TokenServiceImpl;
+import org.mskcc.cbio.oncokb.service.mapper.UserMapper;
+import org.mskcc.cbio.oncokb.web.rest.slack.ActionId;
+import org.mskcc.cbio.oncokb.web.rest.slack.BlockId;
+import org.mskcc.cbio.oncokb.web.rest.slack.DropdownEmailOption;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.context.MessageSource;
+import org.springframework.mail.javamail.JavaMailSenderImpl;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.thymeleaf.spring5.SpringTemplateEngine;
+
+import javax.mail.MessagingException;
+import javax.mail.internet.MimeMessage;
+
+import java.io.IOException;
+import java.net.URL;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.time.Instant;
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.in;
+import static org.mockito.Mockito.*;
+import static org.mskcc.cbio.oncokb.config.Constants.DEFAULT_TOKEN_EXPIRATION_IN_SECONDS;
+
+@SpringBootTest(classes = OncokbPublicApp.class)
+public class SlackControllerIT {
+
+    // Testing Slack url and email addresses for applicationProperties
+    private static final String USER_REGISTRATION_WEBHOOK =  "https://hooks.slack.com/example";
```

Comment:
I see this URL been used multiple times. Is this what slack provided? Or it can be anything

Reply:
It can be anything, is just a test string to ensure the same webhook makes it through to the end of the Slack integration.